c-layer/include/lib.h

```diff
@@ -55,14 +55,55 @@ enum RESOLVABLE_KIND {
 /// @param name resolvable name
 /// @param kind kind of resolvable
 /// @param[out] status (will overwrite existing contents)
+// TODO: add RESOLVABLE_SELECTED value
 void resolvable_select(const char* name, enum RESOLVABLE_KIND kind, struct Status* status) noexcept;
 
 /// Unselect resolvable for installation. It can still be installed as dependency.
 /// @param name resolvable name
 /// @param kind kind of resolvable
 /// @param[out] status (will overwrite existing contents)
+// TODO: add RESOLVABLE_SELECTED value
 void resolvable_unselect(const char* name, enum RESOLVABLE_KIND kind, struct Status* status) noexcept;
 
+enum RESOLVABLE_SELECTED {
+  // resolvable won't be installed
+  NOT_SELECTED,
+  // user select resolvable for installation
+  // match TransactByValue::USER
+  USER_SELECTED,
+  // installation proposal selects resolvable
+  // match TransactByValue::APPL_{LOW,HIGH} we do not need both, so we use just one value
+  INSTALLATION_SELECTED,
+  // dependency solver select resolvable
+  // match TransactByValue::SOLVER
+  SOLVER_SELECTED,
```

Comment:
Reorder these in order of "severity": NOT, SOLVER, INSTALLATION (rename to INSTALLER?), USER

Reply:
well, if I rename, then maybe APPLICATION is the best match to libzypp and can help if we decide to use it elsewhere.